src/app/lib/device/airplay.js

```diff
@@ -1,56 +1,124 @@
 (function (App) {
     'use strict';
 
-    var airplayer = require('airplayer'),
-        netw = require('network-address'),
-        list = airplayer(),
+    /*
+      Useful AirPlay protocol references:
+      - "Unofficial AirPlay Protocol Specification" (https://nto.github.io/AirPlay.html)
+      - @watson's "airplay-protocol" (https://github.com/watson/airplay-protocol)
+    */
+    var Bonjour = require('bonjour'),
+        AirPlayProtocol = require('airplay-protocol');
+
+    var bonjourBrowser = new Bonjour(),
         collection = App.Device.Collection;
 
-    var makeID = function (baseID) {
-        return 'airplay-' + baseID.replace('.', '');
+    var makeID = function (bonjourService) {
+        // strong de-duplication ID since there can only be 1 bonjour service per host+port
+        var uniqueDeviceId = bonjourService.host + '_' + bonjourService.port;
+        return 'airplay-' + Common.md5(uniqueDeviceId);
```

Comment:
Also: Many of the device libraries to string replacement on the device IDs to make them "safe"? Like this:

https://github.com/butterproject/butter-desktop/blob/master/src/app/lib/device/dlna.js#L9

By doing Common.md5 we don't need to clean up the "host/ip_port" string.